### PR TITLE
Fix/update versions

### PR DESCRIPTION
**Pull request by leavauchier**
-  Update pdal version to 2.10 and related fixes:
    
    - update hydra version + bugfixes
    - lower minimum coverage
    - update ign-pdal-tools version (and dependencies)
    - update pytorch version with related fixes

- Add tests to improve coverage (forgotten in a former MR) + lower minimum coverage

### CODE REVIEW

File: tests/myria3d/pctl/dataset/test_iterable.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import numpy as np
+
+from myria3d.pctl.dataset.toy_dataset import TOY_LAS_DATA
```
Comment (yoann-apel) [inline tests/myria3d/pctl/dataset/test_iterable.py:3]: Toy ? C'est pour dire que c'est des données de test ?

**Review by yoann-apel — APPROVED** [review]

File: tests/myria3d/pctl/dataset/test_iterable.py (hunk 1 above)
Comment (leavauchier) [reply tests/myria3d/pctl/dataset/test_iterable.py:3]: c'est parce que c'est n tout petit jeu de test de 100m par 100m utilisé pour les tests oui
